service/src/main/java/bio/terra/cbas/util/methods/GithubUrlDetailsManager.java

```diff
@@ -0,0 +1,84 @@
+package bio.terra.cbas.util.methods;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GithubUrlDetailsManager {
+
+  public GithubUrlComponents extractDetailsFromUrl(String url) throws URISyntaxException {
```

Comment:
small nit: since this is specific to Github maybe can be renamed `extractGithubDetailsFromUrl`?